apm-agent-plugins/apm-es-restclient-plugin/src/main/java/co/elastic/apm/es/restclient/ElasticsearchRestClientInstrumentation.java

```diff
@@ -0,0 +1,147 @@
+        span.getContext().getDb().withType(DB_CONTEXT_TYPE);
+        span.activate();
+
+        if (span.isSampled()) {
+            span.getContext().getHttp().withMethod(request.getMethod());
+            if (request.getEndpoint().endsWith(SEARCH_QUERY_PATH_SUFFIX)) {
+                HttpEntity entity = request.getEntity();
+                if (entity != null && entity.isRepeatable()) {
+                    try {
+                        String body = ESRestClientInstrumentationHelper.readRequestBody(entity.getContent(), request.getEndpoint());
+                        if (body != null && !body.isEmpty()) {
+                            span.getContext().getDb().withStatement(body);
+                        }
+                    } catch (IOException e) {
+                        // We can't log from here
+                    }
+                }
+            }
+        }
+    }
+
+    @Advice.OnMethodExit(onThrowable = Throwable.class)
+    public static void onAfterExecute(@Advice.Return @Nullable Response response,
+                                      @Advice.Local("span") @Nullable Span span,
+                                      @Advice.Thrown @Nullable Throwable t) {
+        if (span != null) {
+            try {
+                String url = null;
+                int statusCode = -1;
+                if(response != null) {
+                    url = response.getHost().toURI().toString();
+                    statusCode = response.getStatusLine().getStatusCode();
+                } else if(t != null) {
+                    if (t instanceof ResponseException) {
+                        ResponseException esre = (ResponseException) t;
+                        url = esre.getResponse().getHost().toURI().toString();
+                        statusCode = esre.getResponse().getStatusLine().getStatusCode();
+
+                        // Add tags so that they will be copied to error capture
+                        span.addTag(QUERY_STATUS_CODE_KEY, Integer.toString(statusCode));
+                        span.addTag(ELASTICSEARCH_NODE_URL_KEY, url);
+                        span.addTag(ERROR_REASON_KEY, esre.getResponse().getStatusLine().getReasonPhrase());
+                    }
+                    span.captureException(t);
+                }
+
+                if(url != null && !url.isEmpty()) {
+                    span.getContext().getHttp().withUrl(url);
+                }
+                if(statusCode > 0) {
+                    span.getContext().getHttp().withStatusCode(statusCode);
+                }
+            } finally {
+                span.deactivate().end();
+            }
+
+        }
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("org.elasticsearch.client.RestClient");
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("performRequest")
```

Comment:
Can be replaces with
```
named("performRequest")
    .and(takesArguments(named("org.elasticsearch.client.Request")))
```

Reply:
No, it can't. Did you mean I can remove the num of arguments check?